### DISCUSSION TITLE
Components seem to not be updated

### DISCUSSION THREAD

**Discussion by trading-peter** [lit/lit#2829] [General]
I have this weird issue where the properties of web components are seemingly not updated when the state comes from within objects. This probably doesn't make any sense to who reads this (lol) so I created a very easy example of what I mean with what I'd expect to happen and what actually happens.

Any hinds to what I'm doing wrong would be very apprechiated.

https://stackblitz.com/edit/lit-html-ytyyqb?file=src/index.js

**Comment by graynorton** [comment]
When rendering, Lit tries (for performance reasons) to avoid updating dynamic template parts whose values have not changed. By default, rather than comparing the current value of the part to the current state of the DOM, it compares the current value of the part to the value of the part *the last time the template was rendered*.

In the case of your repro steps, the value of the `checked` property for the 2nd toggle is being changed by a user action between two rendering passes. In Lit's eyes, the value was `false` in the initial render and is also `false` in the subsequent render, so it doesn't touch the DOM.

In general, it's best to avoid this situation by ensuring that your internal representation of state is kept in sync with the DOM; for example, you might put an event handler in your `<my-app>` component to respond to state changes in the individual toggles and update its own state accordingly.

If for some reason you can't do this, you can use Lit's [`live`](https://lit.dev/docs/templates/directives/#live) directive to make Lit ignore the previously rendered value and instead compare against the current value of the DOM. Here's your example updated to do that (I moved it from Stackblitz to the Lit playground because I find that easier to work with):

https://tinyurl.com/v2ak9zmn

**Reply by trading-peter** [reply]
Oh I understand! Thank you very much for the explanation.
